src/testdir/test_expr.vim

```diff
@@ -946,4 +946,48 @@ func Test_string_interp()
   call v9.CheckDefAndScriptSuccess(lines)
 endfunc
 
+" Test for bitwise left and right shift (<< and >>)
+func Test_bitwise_shift()
+  let lines =<< trim END
+    call assert_equal(16, 1 << 4)
+    call assert_equal(2, 16 >> 3)
+    call assert_equal(0, 0 << 2)
+    call assert_equal(0, 0 >> 4)
+    call assert_equal(3, 3 << 0)
+    call assert_equal(3, 3 >> 0)
+    call assert_equal(0, 0 >> 4)
```

Comment:
What happens if you do:

* `42 >> -1`  (in C, shifting by negative number is undefined behavior, in Vim it should be an error or maybe 0?)
* `42 << 100` or `42 >> 100` (in C, shift by more bits than the underlying type is undefined behavior, in Vim it should be an error or maybe 0?)